ddtrace/tracer/option_test.go

```diff
@@ -249,6 +249,94 @@ func TestServiceName(t *testing.T) {
 	})
 }
 
+func TestTagSeparators(t *testing.T) {
+	assert := assert.New(t)
+
+	t.Run("env-tags", func(t *testing.T) {
+		for ind, tag := range []struct {
+			in       string
+			expected map[string]string
+		}{
+			{
+				in: "env:test aKey:aVal bKey:bVal cKey:",
+				expected: map[string]string{
+					"env":  "test",
+					"aKey": "aVal",
+					"bKey": "bVal",
+					"cKey": ""},
+			},
+			{
+				in: "env:test,aKey:aVal,bKey:bVal,cKey:",
+				expected: map[string]string{
+					"env":  "test",
+					"aKey": "aVal",
+					"bKey": "bVal",
+					"cKey": ""},
+			},
+			{
+				in: "env:test,aKey:aVal bKey:bVal cKey:",
+				expected: map[string]string{
+					"env":  "test",
+					"aKey": "aVal bKey:bVal cKey:"},
+			},
+			{
+				in: "env:test     bKey :bVal dKey: dVal cKey:",
+				expected: map[string]string{
+					"env":  "test",
+					"bKey": "",
+					"dKey": "",
+					"dVal": "",
+					"cKey": ""},
+			},
+			{
+				in: "env :test, aKey : aVal bKey:bVal cKey:",
+				expected: map[string]string{
+					"env":  "test",
+					"aKey": "aVal bKey:bVal cKey:"},
+			},
+			{
+				in: "env:keyWithA:Semicolon bKey:bVal cKey",
+				expected: map[string]string{
+					"env":  "keyWithA:Semicolon",
+					"bKey": "bVal",
+					"cKey": ""},
+			},
+			{
+				in: "env:keyWith:  , ,   Lots:Of:Semicolons ",
+				expected: map[string]string{
+					"env":  "keyWith:",
+					"Lots": "Of:Semicolons"},
+			},
+			{
+				in: "a:b,c,d",
+				expected: map[string]string{
+					"a": "b",
+					"c": "",
+					"d": ""},
+			},
+			{
+				in: "a,1",
+				expected: map[string]string{
+					"a": "",
+					"1": ""},
+			},
+			{
+				in:       "a:b:c:d",
+				expected: map[string]string{"a": "b:c:d"},
+			},
+		} {
+			os.Setenv("DD_TAGS", tag.in)
```

Comment:
```suggestion
			t.Run("", func(t *testing.T) {
				os.Setenv("DD_TAGS", tag.in)
```
Tip: if you do this, with empty string sub-test names, Go will automatically count each test and you'll know which one failed. With this method, you can drop `ind` (which should generally be `i` when used)